plugins/woocommerce/includes/class-wc-brands.php

```diff
@@ -237,10 +237,70 @@ public function add_body_class( $classes ) {
 	 * Enqueues styles.
 	 */
 	public function styles() {
+		if ( ! $this->should_load_brands_styles() ) {
+			return;
+		}
+
 		$version = Constants::get_constant( 'WC_VERSION' );
 		wp_enqueue_style( 'brands-styles', WC()->plugin_url() . '/assets/css/brands.css', array(), $version );
 	}
 
+	/**
+	 * Determines if brands styles should be loaded on the current page.
+	 *
+	 * @since 10.4.0
+	 * @return bool
+	 */
+	private function should_load_brands_styles() {
+		global $post;
+
+		// Should load on brand taxonomy archive pages.
+		if ( is_tax( 'product_brand' ) ) {
+			return true;
+		}
+
+		// Should load on single product pages that have brands assigned.
+		if ( is_singular( 'product' ) && $post ) {
+			$terms = get_the_terms( $post->ID, 'product_brand' );
+			if ( ! empty( $terms ) && ! is_wp_error( $terms ) ) {
+				return true;
+			}
+		}
```

Comment:
Can we simplify a bit if we use `has_term()`?

```suggestion
		if ( is_singular( 'product' ) && has_term( '', 'product_brand' ) ) {
			return true;
		}
```

Reply:
Woops, much better 😅  I think I defaulted to the pattern I saw elsewhere in the file, but that's used for different reasons.